# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Mukeshs-06/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

**Chosen Lane:** Refresh / Content Opportunity Scoring

**Why this lane:** Content performance across large publishing ecosystems naturally degrades over time as search intent evolves, competitors publish newer material, and information becomes outdated. Manually reviewing thousands of published articles is unfeasible for editorial teams. I chose the **Refresh / Content Opportunity Scoring** lane because prioritizing high-leverage content updates directly impacts organic traffic retention. By ranking pages that exhibit observable signals of decay, we can direct editorial resources to where content refreshes offer the highest return on investment.

In [1]:
# Section 1 summary check
print("Selected Lane: Refresh / Content Opportunity Scoring")


Selected Lane: Refresh / Content Opportunity Scoring


## 2. The question: decision, action, cost of a wrong call

- **Decision improved:** Which declining or underperforming content items should an editorial team prioritize updating first to mitigate traffic loss?
- **Who acts and what they do:** Content editors and SEO strategists. They inspect prioritized pages, update outdated information, refresh keywords/metadata, or restructure content to restore search relevance.
- **Cost of a wrong call:**
  - *False Positive (flagging a healthy page):* Wastes ~3–5 hours of editor bandwidth and budget on unnecessary content rewrites.
  - *False Negative (missing a severely declining page):* Leads to compounding organic search traffic loss and reduced audience engagement before intervention occurs.
- **Why data/ML helps:** A simple rule (e.g., "refresh every page older than 12 months") ignores multi-dimensional search dynamics. Machine learning enables ranking pages by combining multiple non-linear signals (impressions, CTR trends, position shifts, and engagement rates) into an explainable priority score.

In [2]:
# Section 2 verification
print("Framing complete: Decision, Actor/Action, Error Costs, and ML Rationale defined.")


Framing complete: Decision, Actor/Action, Error Costs, and ML Rationale defined.


## 3. Quick look at the data (2-3 real numbers)

Below, we load the starter dataset (`data/raw/content_refresh_anonymized.csv`) and compute key metrics that illustrate the scale of content decay within this dataset.

In [3]:
import pandas as pd

# Load starter dataset relative to work/notebooks/
df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

# Calculate real numbers
total_pages = len(df)
declining_pages = (df["trend_direction"] == "down").sum()
pct_declining = (declining_pages / total_pages) * 100
total_clicks = df["clicks_90d"].sum()
declining_clicks = df[df["trend_direction"] == "down"]["clicks_90d"].sum()
pct_declining_clicks = (declining_clicks / total_clicks) * 100

print(f"Total content items evaluated: {total_pages:,}")
print(f"Declining content items (trend_direction == 'down'): {declining_pages:,} ({pct_declining:.2f}%)")
print(f"Total 90-day clicks across dataset: {total_clicks:,}")
print(f"Clicks from declining pages: {declining_clicks:,} ({pct_declining_clicks:.2f}%)")


Total content items evaluated: 30,000
Declining content items (trend_direction == 'down'): 16,262 (54.21%)
Total 90-day clicks across dataset: 482,920
Clicks from declining pages: 217,248 (44.99%)


## 4. Careful words: what I can and can't claim

- **What can be claimed:**
  - *Observed performance:* Within this dataset of 30,000 pages across 32 clients, 54.21% of pages are in a downward traffic trend.
  - *Directional patterns:* We can measure statistical correlations between feature changes (e.g., impression trends, average position shifts) and traffic decline.
  - *Decision-support utility:* We can produce a ranked recommendation queue to help editors prioritize content review.
- **What CANNOT be claimed:**
  - *Causal proof:* We cannot claim that updating a page will guarantee ranking recovery or specific click gains.
  - *Algorithm prediction:* We do not claim to predict or reverse-engineer Google's search algorithm.

In [4]:
# Section 4 claim boundary check
print("Claim boundaries established: Observed, Directional, Decision-Support only.")


Claim boundaries established: Observed, Directional, Decision-Support only.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.